vultr/provider_test.go

```diff
@@ -1,16 +1,35 @@
 package vultr
 
 import (
+	"os"
 	"testing"
 
 	"github.com/hashicorp/terraform/helper/schema"
 	"github.com/hashicorp/terraform/terraform"
 )
 
-var _ terraform.ResourceProvider = Provider()
+var testAccProviders map[string]terraform.ResourceProvider
+var testAccProvider *schema.Provider
+
+func init() {
+	testAccProvider = Provider().(*schema.Provider)
+	testAccProviders = map[string]terraform.ResourceProvider{
+		"vultr": testAccProvider,
+	}
+}
 
 func TestProvider(t *testing.T) {
 	if err := Provider().(*schema.Provider).InternalValidate(); err != nil {
-		t.Fatalf("expected provider to validate: %v", err)
+		t.Fatalf("err: %s", err)
+	}
+}
+
+func TestProvider_impl(t *testing.T) {
+	var _ terraform.ResourceProvider = Provider()
+}
+
+func testAccPreCheck(t *testing.T) {
+	if v := os.Getenv("VULTR_API_KEY"); v == "" {
```

Comment:
There's no need to store this in a variable since we're not using `v` for anything. we can say:
```go
if os.Getenv("VULTR_API_KEY") == "" {
...
}
```